library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2154,17 +1856,26 @@ impl<T, A: Allocator> VecDeque<T, A> {
     #[inline]
     #[stable(feature = "append", since = "1.4.0")]
     pub fn append(&mut self, other: &mut Self) {
-        self.reserve(other.len());
+        if T::IS_ZST {
+            self.len += other.len;
+            other.len = 0;
+            other.head = 0;
+            return;
+        }
+
+        self.reserve(other.len);
         unsafe {
             let (left, right) = other.as_slices();
-            self.copy_slice(self.head, left);
-            self.copy_slice(self.wrap_add(self.head, left.len()), right);
+            self.copy_slice(self.wrap_idx(self.len), left);
+            // no overflow, because self.capacity() >= old_cap + left.len() >= self.len + left.len()
+            self.copy_slice(self.wrap_idx(self.len + left.len()), right);
         }
         // SAFETY: Update pointers after copying to avoid leaving doppelganger
         // in case of panics.
-        self.head = self.wrap_add(self.head, other.len());
+        self.len += other.len;
         // Silently drop values in `other`.
```

Comment:
this isn't your comment, but while you're here would you mind improving this comment a bit?  It's definitely not *drop*ing the values in `other`, as that would be unsound, so maybe something like
```suggestion
        // Now that we own its values, forget everything in `other`.
```